<launch>
    <arg name="veh" doc="Name of vehicle. ex: megaman"/>
    <arg name="local" default="true" doc="true for running everything you possibly can on laptop. false for everything running on vehicle."/>
    <arg name="config" default="baseline" doc="Specify a config."/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman." />

    <arg name="pkg_name" value="virtual_mirror_wubella" />
    <arg name="node_name" default="virtual_mirror_node" />
    
    <node pkg="$(arg pkg_name)" name="$(arg node_name)" type="$(arg node_name).py" output="screen">
        <rosparam command="load" file="$(find duckietown_wubella)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
    </node>

    <group ns="$(arg veh)">        
        <!-- run local -->
        <node if="$(arg local)" name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen">
            <rosparam command="load" file="$(find duckietown_wubella)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
        </node>

        <!-- run remote -->
        <node unless="$(arg local)" machine="$(arg veh)" name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen">
            <rosparam command="load" file="$(find duckietown_wubella)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
        </node>
    </group>
</launch>